cmd/podman/containers/export.go

```diff
@@ -67,14 +68,24 @@ func init() {
 }
 
 func export(cmd *cobra.Command, args []string) error {
-	if len(exportOpts.Output) == 0 {
+	if len(outputFile) == 0 {
 		file := os.Stdout
 		if term.IsTerminal(int(file.Fd())) {
 			return errors.New("refusing to export to terminal. Use -o flag or redirect")
 		}
-		exportOpts.Output = "/dev/stdout"
-	} else if err := parse.ValidateFileName(exportOpts.Output); err != nil {
-		return err
+		exportOpts.Output = file
+	} else {
+		if err := parse.ValidateFileName(outputFile); err != nil {
+			return err
+		}
+		// open file here with WRONLY since on MacOS it can fail to open /dev/stderr in read mode for example
+		// https://github.com/containers/podman/issues/16870
+		file, err := os.OpenFile(outputFile, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
```

Comment:
since we are at it, wouldn't it make sense to restrict the file access to the owner (mode: 0600)?

Reply:
Maybe, the old os.Create()  used `0666`. However this could be considered a breaking change if users rely on it.